### PR TITLE
chore(root): Release npm packages

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?

- [x] Update to latest NX
- [x] Release using latest nx release command.

Learn more at:
https://nx.dev/features/manage-releases
https://nx.dev/nx-api/nx/documents/release

ℹ️ *Important note* ℹ️

Due to [a bug in NX](https://github.com/nrwl/nx/issues/28996), you cannot use `nx publish --otp` and publish all packages at once with a single command. As a result, we have to run nx release subcommands separately.

### CODE REVIEW

File: apps/api/package.json [hunk 1]
```diff
@@ -99,8 +99,7 @@
     "twilio": "^4.14.1",
     "uuid": "^8.3.2",
     "zod": "^3.23.8",
-    "zod-to-json-schema": "^3.23.3",
-    "json-schema-to-ts": "^3.0.0"
```
Comment (rifont) [inline apps/api/package.json:103]: This dependency (`json-schema-to-ts`) got removed, just confirming it's intentional?

File: nx.json [hunk 2]
```diff
@@ -16,6 +16,32 @@
...
+      "@novu/headless",
+      "@novu/js",
+      "@novu/react",
+      "@novu/react-native",
+      "@novu/nextjs",
+      "@novu/node",
+      "@novu/providers",
+      "@novu/shared",
+      "@novu/stateless"
+    ],
```
Comment (rifont) [inline nx.json:37]: 🦾

File: package.json [hunk 3]
```diff
@@ -187,7 +181,7 @@
     "typescript": "5.6.2",
     "typescript-eslint": "^8.3.0",
     "wait-port": "^0.3.0",
-    "nx-cloud": "latest"
+    "nx-cloud": "19.1.0"
```
Comment (rifont) [inline package.json:184]: Is it intended for `nx-cloud` to be 1 major version behind?

File: package.json [hunk 4]
```diff
@@ -223,7 +217,7 @@
     "pnpm": "^9.11.0"
   },
   "dependencies": {
-    "nx": "19.8.2",
+    "nx": "20.1.2",
```
Comment (rifont) [inline package.json]: Nit: `nx` should be in `devDependencies`

File: apps/api/package.json (hunk 1 above)
Comment (SokratisVidros) [reply apps/api/package.json:103]: Double entry: https://github.com/novuhq/novu/pull/7073/files#diff-2c40985d6d91eed8ae85ec1c8e754a85984ee32e156a600d2b7a467423d7e338L77

File: package.json (hunk 3 above)
Comment (SokratisVidros) [reply package.json:184]: Yes, they haven't released a 20.X yet. PTAL https://www.npmjs.com/package/nx-cloud

File: package.json (hunk 4 above)
Comment (SokratisVidros) [reply package.json]: Oops, duplicate entry. Removed.

File: packages/novu/package.json [hunk 5]
```diff
@@ -1,10 +1,9 @@
...
   "name": "novu",
-  "version": "2.2.0-rc.2",
+  "version": "2.2.0",
   "description": "Novu CLI. Run Novu Studio and sync workflows with Novu Cloud",
   "main": "src/index.js",
   "publishConfig": {
-    "registry": "https://registry.npmjs.org",
     "access": "public"
   },
   "private": false,
```
Comment (rifont) [inline packages/novu/package.json:9]: chore: see `@novu/shared` dependency in `dependencies`, it should be udpated to use `workspace:*` dependency (got accidentally bumped in the last release)

**Review by rifont — APPROVED** [review]

File: packages/novu/package.json (hunk 5 above)
Comment (SokratisVidros) [reply packages/novu/package.json:9]: Fixed.

File: apps/dashboard/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@novu/dashboard",
   "private": true,
-  "version": "0.0.0",
+  "version": "0.0.1",
```
Comment (LetItRock) [inline apps/dashboard/package.json:4]: is there an option to skip updating the private packages?

Comment (SokratisVidros) [reply apps/dashboard/package.json:4]: I will have a look.

**Pull request merged**